packages/composites/bob/src/endpoint/format.ts

```diff
@@ -0,0 +1,39 @@
+import JSONRPC from '@chainlink/json-rpc-adapter'
+import { ExecuteWithConfig } from '@chainlink/types'
+import { Validator, Requester } from '@chainlink/ea-bootstrap'
+import { DEFAULT_RPC_URL, ExtendedConfig } from '../config'
+import { ethers } from 'ethers'
+
+export const NAME = 'format'
+
+export const inputParams = {
+    url: false,
+    chainId: true,
+    blockNumber: true
+}
+
+
+export const execute: ExecuteWithConfig<ExtendedConfig> = async (request, config) => {
+    const validator = new Validator(request, inputParams)
+    if (validator.error) throw validator.error
+    const url = config.RPC_URL || validator.validated.data.url || DEFAULT_RPC_URL
```

Comment:
Not sure if this will work as intended.

`RPC_URL` is a required environment parameter. The EA will error out without it.
When `RPC_URL` is supplied this line's logical OR operators will check from the left. Since `RPC_URL` is always there it will never hit the right two cases.

One way to change this would be to use `util.getEnv` in `config.ts` (and update the README with RPC_URL now being optional).